### PR TITLE
clang-tidy: modernize replace auto_ptr

### PR DESCRIPTION
**Pull request by est77**
Automatic changes, using clang-tidy -fix, with only minor manual fixes.

It seems gcc 4.8 has issues with unqualified move calls in some specific cases. Some std:: qualifiers are needed for the code to compile.

Some new headers are also needed to avoid errors when using unique_ptr with incomplete classes.

### CODE REVIEW

File: src/appleseed.python/bindmasterrenderer.cpp [hunk 1]
```diff
@@ -50,8 +51,10 @@ namespace
     // A class that wraps MasterRenderer and keeps a Python
     // reference to the project object to prevent it being
     // destroyed by Python before the MasterRenderer is destroyed.
-    struct MasterRendererWrapper
+    class MasterRendererWrapper
+      : NonCopyable
```
Comment (dictoon) [inline src/appleseed.python/bindmasterrenderer.cpp]: Inherit publicly?

File: src/appleseed.python/bindmasterrenderer.cpp [hunk 2]
```diff
@@ -85,35 +88,33 @@ namespace
         }
 
         bpy::object                     m_project;
-        std::auto_ptr<MasterRenderer>   m_renderer;
+        std::unique_ptr<MasterRenderer> m_renderer;
     };
 
-    std::auto_ptr<MasterRendererWrapper> create_master_renderer(
+    std::shared_ptr<MasterRendererWrapper> create_master_renderer(
```
Comment (dictoon) [inline src/appleseed.python/bindmasterrenderer.cpp]: Why `shared_ptr<>` here?

File: src/appleseed.studio/mainwindow/project/attributeeditor.cpp [hunk 3]
```diff
@@ -36,6 +36,7 @@
 
 // Qt headers.
 #include <QLayout>
+#include <utility>
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/project/attributeeditor.cpp]: Not a Qt header.

File: src/appleseed.studio/mainwindow/project/disneymateriallayerui.cpp [hunk 4]
```diff
@@ -185,19 +186,19 @@ void DisneyMaterialLayerUI::create_input_widgets(const Dictionary& values)
...
                     : create_colormap_input_widgets(im) :
             input_type == "text" ? create_text_input_widgets(im) :
-            auto_ptr<IInputWidgetProxy>(0);
+            unique_ptr<IInputWidgetProxy>(nullptr);
 
         assert(widget_proxy.get());
         connect(widget_proxy.get(), SIGNAL(signal_changed()), SIGNAL(signal_apply()));
 
-        m_widget_proxies.insert(input_name, widget_proxy);
+        m_widget_proxies.insert(input_name, std::move(widget_proxy));
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/project/disneymateriallayerui.cpp]: `std::` qualifier redundant? (In many other places as well.)

File: src/appleseed.studio/mainwindow/mainwindow.cpp [hunk 5]
```diff
@@ -1736,11 +1736,11 @@ namespace
 
 void MainWindow::slot_clear_render_region()
 {
-    auto_ptr<RenderingManager::IScheduledAction> clear_render_region_action(
+    unique_ptr<RenderingManager::IScheduledAction> clear_render_region_action(
         new ClearRenderRegionAction(m_attribute_editor));
 
     if (m_rendering_manager.is_rendering())
-        m_rendering_manager.schedule(clear_render_region_action);
+        m_rendering_manager.schedule(std::move(clear_render_region_action));
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/mainwindow.cpp]: `std::` qualifier redundant?

File: src/appleseed.studio/mainwindow/mainwindow.cpp [hunk 6]
```diff
@@ -1762,7 +1762,7 @@ void MainWindow::slot_set_render_region(const QRect& rect)
     }
     else
     {
-        m_rendering_manager.schedule(set_render_region_action);
+        m_rendering_manager.schedule(std::move(set_render_region_action));
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/mainwindow.cpp]: `std::` qualifier redundant?

**Comment by dictoon** [comment]
Builds with VS 2012 so good to merge once review comments have been addressed!

Then we can add a comment to https://github.com/appleseedhq/appleseed/issues/93 that the replacement has finally been done.

File: src/appleseed.studio/mainwindow/project/disneymateriallayerui.cpp (hunk 4 above)
Comment (est77) [reply src/appleseed.studio/mainwindow/project/disneymateriallayerui.cpp]: These qualifiers are needed for gcc 4.8 to compile the code.
Some libraries define move methods and functions, for example Qt and the compiler does not pick std::move.

File: src/appleseed.python/bindmasterrenderer.cpp (hunk 2 above)
Comment (est77) [reply src/appleseed.python/bindmasterrenderer.cpp]: We cannot pass unique_ptr to bpy::make_constructor because unique_ptr is non-copyable.

**Comment by est77** [comment]
Fixed and rebased

**Pull request merged**